### PR TITLE
Fix the the request logging boolean toggle

### PR DESCRIPTION
**Pull request by vagababov**
Since we have cut the 0.17, we can go for the proper toggle setup:

- if request logging is on — template must be present and the logging will be on
- if the request of logging is off — template can be present, but will not affect whether the request logs will be collected

Really
Fixes https://github.com/knative/serving/issues/8644

/assign @yanweiguo @julz

### CODE REVIEW

**Comment by vagababov** [comment]
(ノಠ益ಠ)ノ彡┻━┻

Unrelated
/retest

File: metrics/config_observability.go [hunk 1]
```diff
@@ -92,15 +95,8 @@ func NewObservabilityConfigFromConfigMap(configMap *corev1.ConfigMap) (*Observab
...
-	if raw, ok := configMap.Data["logging.enable-request-log"]; ok {
-		if strings.EqualFold(raw, "true") && oc.RequestLogTemplate != "" {
-			oc.EnableRequestLog = true
-		}
-	} else if oc.RequestLogTemplate != "" {
-		// TODO: remove this after 0.17 cuts, this is meant only for smooth transition to the new flag.
-		// Once 0.17 cuts we should set a proper default value and users will need to set the flag explicitly
-		// to enable request logging.
-		oc.EnableRequestLog = true
+	if oc.RequestLogTemplate == "" && oc.EnableRequestLog {
```
Comment (yanweiguo) [inline metrics/config_observability.go:98]: I think the idea is to use the default value which set in `_example` in the configmap?

Comment (vagababov) [reply metrics/config_observability.go:98]: I thought to do this after importing to Serving. But I guess, we can just do it.

Comment (vagababov) [reply metrics/config_observability.go:101]: Done.

File: metrics/config_observability_test.go [hunk 2]
```diff
@@ -49,38 +48,47 @@ func TestObservabilityConfiguration(t *testing.T) {
...
 			"logging.enable-probe-request-log":            "true",
+			"logging.enable-request-log":                  "false",
 			"logging.enable-var-log-collection":           "true",
+			"logging.request-log-template":                "",
+			"logging.revision-url-template":               "https://logging.io",
+			"metrics.request-metrics-backend-destination": "stackdriver",
+			"profiling.enable":                            "true",
+		},
+	}, {
+		name:    "observability configuration with request log set and template not set",
```
Comment (yanweiguo) [inline metrics/config_observability_test.go:84]: My understanding is that if request log set but template not set, we use default tempalte. If template set to empty, throw an error. Is my understanding wrong?

Comment (vagababov) [reply metrics/config_observability_test.go:84]: yep that's why it's zeroed out in the test.

**Pull request merged**